clients/client/src/test/java/org/projectnessie/model/TestModelObjectsSerialization.java

```diff
@@ -0,0 +1,202 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.projectnessie.model.Contents.Type;
+
+/**
+ * This test merely checks the JSON serialization/deserialization of the model classes, with an
+ * intention to identify breaking cases whenever jackson version varies.
+ */
+public class TestModelObjectsSerialization {
+
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static final String HASH =
+      "3e1cfa82b035c26cbbbdae632cea070514eb8b773f616aaeaf668e2f0be8f10e";
+
+  @ParameterizedTest
+  @MethodSource("goodCases")
+  void testGoodSerDeCases(Case goodCase) throws IOException {
+    String json = MAPPER.writeValueAsString(goodCase.obj);
+    Assertions.assertThat(json).isEqualTo(goodCase.deserializedJson);
+    Object deserialized = MAPPER.readValue(json, goodCase.deserializeAs);
+    Assertions.assertThat(deserialized).isEqualTo(goodCase.obj);
+  }
+
+  @ParameterizedTest
+  @MethodSource("negativeCases")
+  void testNegativeSerDeCases(Case invalidCase) {
+    assertThrows(
+        JsonProcessingException.class,
+        () -> MAPPER.readValue(invalidCase.deserializedJson, invalidCase.deserializeAs));
+  }
+
+  static List<Case> goodCases() {
+    final Instant now = Instant.now();
+    final String branchName = "testBranch";
+
+    return Arrays.asList(
+        new Case(
+            ImmutableBranch.builder().hash(HASH).name(branchName).build(),
+            Branch.class,
+            "{\"type\":\"BRANCH\",\"name\":\"testBranch\"," + "\"hash\":\"" + HASH + "\"}"),
+        new Case(
+            ImmutableBranch.builder().name(branchName).build(),
+            Branch.class,
+            "{\"type\":\"BRANCH\",\"name\":\"testBranch\",\"hash\":null}"),
+        new Case(
+            ImmutableTag.builder().hash(HASH).name("tagname").build(),
+            Tag.class,
+            "{\"type\":\"TAG\",\"name\":\"tagname\",\"hash\":\"" + HASH + "\"}"),
+        new Case(
+            ImmutableHash.builder().name(HASH).build(),
+            Hash.class,
+            "{\"type\":\"HASH\",\"name\":\"" + HASH + "\",\"hash\":\"" + HASH + "\"}"),
+        new Case(
+            ImmutableTransplant.builder()
+                .addHashesToTransplant(HASH)
+                .fromRefName(branchName)
+                .build(),
+            Transplant.class,
+            "{\"hashesToTransplant\":" + "[\"" + HASH + "\"]," + "\"fromRefName\":\"testBranch\"}"),
+        new Case(
+            ImmutableEntriesResponse.builder()
+                .addEntries(
+                    ImmutableEntry.builder()
+                        .type(Type.ICEBERG_TABLE)
+                        .name(ContentsKey.fromPathString("/tmp/testpath"))
+                        .build())
+                .token(HASH)
+                .hasMore(true)
+                .build(),
+            EntriesResponse.class,
+            "{\"hasMore\":true,"
```

Comment:
I think, this can be more written more compact, especially when using `String.format()` for `HASH` and `now`.